src/Database/DatabaseConfig.php

```diff
@@ -97,11 +97,24 @@ public function makeCredentials(): void
         }
     }
 
-    public function getTemplateConnectionName(): string
+    public function getTemplateConnectionDriver(): string
     {
-        return $this->tenant->getInternal('db_connection')
-            ?? config('tenancy.database.template_tenant_connection')
-            ?? config('tenancy.database.central_connection');
+        return $this->getTemplateConnection()['driver'];
+    }
+
+    public function getTemplateConnection(): array
+    {
+        if ($template = $this->tenant->getInternal('db_connection')) {
+            return config("database.connections.{$template}");
+        }
+
+        if ($template = config('tenancy.database.template_tenant_connection')) {
+            return is_array($template) ? $template : config("database.connections.{$template}");
```

Comment:
I'm thinking that it should be possible to define the template connection only partially and use the rest from the central connection. So it'd essentially be `array_merge($template, $centralConnection)`.

However, that might lead to some issues e.g. when the central connection has a `url` defined and we don't want to merge that with the central connection.

But maybe in those scenarios the values could be explicitly defined in the template, e.g. `'url' => null`.

If we use the `array_merge()` I suggested above, the central connection has a `url`, and the template has `'url' => null`, will the resulting connection have `'url' => null`?

Reply:
> will the resulting connection have 'url' => null?

No. The result will be `url` not `null`. This is because `array_merge` merges the second array into first array. For example
```php
array_merge(['url' => null], ['url' => 'example.com'])

// output 
[
  "url" => "example.com",
]
```

Now if we switch sides: 
```php
array_merge(['url' => 'example.com'], ['url' => null]);

// output 
[
  "url" => null,
]
```

I can explore/write tests for these and implement them this way if it works as expected. With the second approach, we need to define every field in the `$template` we don't want to use from `$centralConnection`.

Reply:
Implemented the ability to define the connection partial configuration. `array_merge($centralConnection, $template)` works as expected. Added a test and update the PR description.

Reply:
Right, I meant opposite — `array_merge($centralConnection, $template)`. I typed that part of the code wrong but it should be clear what I meant from the rest of the comment.